packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -21,8 +21,14 @@ import { startRum } from './rum'
 export interface RumUserConfiguration extends UserConfiguration {
   applicationId: string
   beforeSend?: (event: RumEvent) => void
+  onNewLocation?: NewLocationListener
```

Comment:
What about putting it behind a flag first?
it could allow us to do some tests internally before releasing it